pkg/languagedetection/detector.go

```diff
@@ -0,0 +1,90 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package languagedetection
+
+import (
+	"strings"
+
+	"github.com/DataDog/datadog-agent/pkg/process/procutil"
+)
+
+// LanguageName is a string enum that represents a detected language name.
+type LanguageName string
+
+const (
+	Node    LanguageName = "node"
+	Dotnet  LanguageName = "dotnet"
+	Python  LanguageName = "python"
+	Java    LanguageName = "java"
+	Unknown LanguageName = ""
+)
+
+// Language contains metadata collected from the call to `DetectLanguage`
+type Language struct {
+	Name LanguageName
+}
```

Comment:
There is meta information which is necessary for properly instrumenting the application. For ex: for java we need to know if this is run via maven or buildpack. See 
https://github.com/DataDog/auto_inject/blob/8bdfdcbfb651435cf3212d6a63c1045cfc459216/preload_go/language_processors.go#L412